tests/api2/test_virt_002_instance.py

```diff
@@ -0,0 +1,153 @@
+from threading import Event
+
+from middlewared.test.integration.assets.filesystem import mkfile
+from middlewared.test.integration.assets.pool import dataset
+from middlewared.test.integration.utils.client import client
+from middlewared.test.integration.utils.call import call
+from middlewared.test.integration.utils.ssh import ssh
+
+from auto_config import pool_name
+
+
+def clean():
+    call('virt.global.update', {'pool': ''}, job=True)
+    ssh(f'zfs destroy -r {pool_name}/.ix-virt || true')
+    call('virt.global.update', {'pool': 'tank'}, job=True)
+
+
+def test_virt_instances_create():
+    clean()
+
+    wait_agent = Event()
+
+    def wait_debian(*args, **kwargs):
+        wait_agent.set()
+
+    with client() as c:
```

Comment:
Should we perhaps have a fixture for this which just ensures that necessary bits are in place before we proceed with the tests?

Reply:
Not sure I am following, isn't this pretty much what clean() is doing?
Or do you have an example somewhere on how it should be?